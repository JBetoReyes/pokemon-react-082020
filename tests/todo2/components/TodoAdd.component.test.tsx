import TodoAdd from '@components/hooks/Todo2/TodoAdd.component';
import React from 'react';
import { shallow } from 'enzyme';
import { AppSubmitEvent } from '@typings/htmlEvents';

describe('TodoAdd', () => {
  const handleAdd = jest.fn();
  const todoAddJsx = <TodoAdd handleAdd={handleAdd} />;
  let wrapper = shallow(todoAddJsx);
  beforeEach(() => {
    wrapper = shallow(todoAddJsx);
  });
  test('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Should call the handle add', () => {
    wrapper.find('input').simulate('change', {
      target: { value: 'test todo1', name: 'description' },
    });
    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({ preventDefault() {} } as AppSubmitEvent);
    expect(handleAdd).toHaveBeenCalled();
    expect(handleAdd).toHaveBeenCalledWith({
      id: expect.any(Number),
      description: 'Test todo1',
      done: false,
    });
    const value = wrapper.find('input').prop('value');
    expect(value).toBe('');
  });
});
