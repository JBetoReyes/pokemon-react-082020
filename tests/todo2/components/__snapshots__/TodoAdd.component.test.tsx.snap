// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoAdd Should match the snapshot 1`] = `
<Fragment>
  <h2>
    Add Todo
  </h2>
  <hr />
  <form
    className="form-group"
    onSubmit={[Function]}
  >
    <input
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Enter a TODO"
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary btn-block mt-3"
      type="submit"
    >
      Add
    </button>
  </form>
</Fragment>
`;
